.github/workflows/build.yml

```diff
@@ -1,25 +1,39 @@
 name: .NET Core CI
 
-on: [push, pull_request]
+on:
+  push:
+    paths-ignore:
+      - '.github/*'
+      - '.github/*_TEMPLATE/**'
+      - '*.md'
+  pull_request:
+    paths-ignore:
+      - '.github/*'
+      - '.github/*_TEMPLATE/**'
+      - '*.md'
+  workflow_dispatch:
 
 jobs:
   build:
-    name: .NET on ${{ matrix.os }}
-
+    name: .NET on ${{ matrix.os }} (${{ matrix.configuration }})
+    runs-on: ${{ matrix.os }}
     strategy:
+      fail-fast: false
       matrix:
-        os: [ubuntu-latest, windows-latest]
-
-    runs-on: ${{ matrix.os }}
-
+        os: [macos-latest, ubuntu-latest, windows-latest]
+        configuration: [Release, Debug]
+    env:
+      DOTNET_CLI_TELEMETRY_OPTOUT: 1
```

Comment:
I'd rather let them gather telemetry here. There's no privacy to be had by a robot in a clean container building publicly available source code...